queries.sql

```diff
@@ -96,4 +96,62 @@ SELECT o.id, o.full_name, COUNT(a.id) as number_of_animals
 FROM owners o
 LEFT JOIN animals a ON o.id = a.owner_id
 GROUP BY o.id
-ORDER BY number_of_animals DESC LIMIT 1;
\ No newline at end of file
+ORDER BY number_of_animals DESC LIMIT 1;
+
+-------------------------------------------------------------
+-- Who was the last animal seen by William Tatcher?
+SELECT animals.name FROM animals 
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'William Tatcher' ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+---------------------------------------------------------------
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(animals.id) FROM animals 
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez';
+-------------------------------------------------------
+-- List all vets and their specialties, including vets with no specialties.
+SELECT vets.name as Vet_name, species.name FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON specializations.species_id = species.id;
+---------------------------------------------------------------
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name FROM animals 
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'Stephanie Mendez' AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+----------------------------------------------------------
+-- What animal has the most visits to vets?
+SELECT animals.name, COUNT(visits.animal_id) as count FROM animals 
+JOIN visits ON animals.id = visits.animal_id
+GROUP BY animals.name
+ORDER BY COUNT(visits.animal_id) DESC LIMIT 1;
+------------------------------------------------------------
+-- Who was Maisy Smith's first visit?
+SELECT animals.name, visits.date_of_visit FROM animals
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets on visits.vet_id = vets.id
+WHERE vets.name = 'Maisy Smith' ORDER BY visits.date_of_visit DESC LIMIT 1;
+------------------------------------------------------------
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT animals.name as animal, vets.name as vet, visits.date_of_visit FROM animals
+JOIN visits ON animals.id = visits.animal_id
+JOIN vets on visits.vet_id = vets.id
+ORDER BY visits.date_of_visit DESC LIMIT 1;
+------------------------------------------------------------
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) FROM visits
+JOIN animals on visits.animal_id = animals.id 
+JOIN vets on visits.vet_id = vets.id
+LEFT JOIN specializations on vets.id = specializations.vet_id 
+WHERE specializations.species_id IS NULL;
+-----------------------------------------------------------
```

Comment:
- [ ] The result for this query is wrong. There are 12 visits in total to vets that did not specialize in that animal's species. The breakdown is as follows:

- Maisy Smith - 9
- William Tatcher - 2
- Jack Harkness - 1

Kindly rewrite this query so that you can get the correct result.

#### N.B: Please do not forget to update the screenshot in the pull request when you get the correct result. 👍 😎

Reply:
Yeah , thank you for the note @Gambit142  😆